test/e2e/admin_test.go

```diff
@@ -8,34 +8,23 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-
 	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/tools/clientcmd"
 )
 
-type testClient struct {
-	kc *kubernetes.Clientset
-}
-
-var c testClient
-
 var _ = BeforeSuite(func() {
-
-	// use the current context in kubeconfig
-	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
-	Expect(err).NotTo(HaveOccurred())
-
-	// create the clientset
-	clientset, err := kubernetes.NewForConfig(config)
-	Expect(err).NotTo(HaveOccurred())
-	c.kc = clientset
+	c = newTestClient(*kubeconfig)
 })
 
-var _ = Describe("Openshift on Azure e2e tests", func() {
+var _ = Describe("Openshift on Azure admin e2e tests", func() {
+	defer GinkgoRecover()
+
 	It("should label nodes correctly", func() {
+		if !c.isAzureClusterReader() {
```

Comment:
I hope we can find a better way to filter these (maybe use tags in test names and run ginkgo by ourselves?). Not sure if it's a blocker for this PR.

Reply:
+1 for tags and use --focus and --skip if we need to.

Reply:
Updated to use tags, ptal